src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -955,6 +959,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
+                                     {:recursive true})
+      (watch-protocol/add-callback!
+       watcher
+       (fn [events]
+         (when (some #(and
+                       (:changed-path %)
+                       (= (.getCanonicalPath (:changed-path %))
+                          normalized-crl-path))
+                     events)
+           (.reload ssl-context-factory)))))))
```

Comment:
Wonder if it's possible for the reload to see a partially written file and crash, or end up with incorrect data.